docs/bluetooth.md

```diff
@@ -0,0 +1,208 @@
+
+### Subscribing to Bluetooth discoveries
+
+Some integrations may need to know when a device is discovered right away. The Bluetooth integration provides a registration API to receive callbacks when a new device is discovered that matches specific key values. The same format for `bluetooth` in [`manifest.json`](creating_integration_manifest.md#bluetooth) is used for matching. In addition to the matchers used in the `manifest.json`, `address` can also be used as a matcher.
+
+The function `bluetooth.async_register_callback` is provided to enable this ability. The function returns a callback that will cancel the registration when called.
+
+The below example shows registering to get callbacks when a Switchbot device is nearby.
+
+```python
+from homeassistant.components import bluetooth
+
+...
+
+@callback
+def _async_discovered_device(service_info: bluetooth.BluetoothServiceInfoBleak, change: bluetooth.BluetoothChange) -> None:
+    """Subscribe to bluetooth changes."""
+    _LOGGER.warning("New service_info: %s", service_info)
+
+entry.async_on_unload(
+    bluetooth.async_register_callback(
+        hass, _async_discovered_device, {"service_uuid": "cba20d00-224d-11e6-9fb8-0002a5d5c51b", "connectable": False}, bluetooth.BluetoothScanningMode.ACTIVE
+    )
+)
+```
+
+The below example shows registering to get callbacks for HomeKit devices.
+
+```python
+from homeassistant.components import bluetooth
+
+...
+
+entry.async_on_unload(
+    bluetooth.async_register_callback(
+        hass, _async_discovered_homekit_device, {"manufacturer_id": 76, "manufacturer_data_first_byte": 6}, bluetooth.BluetoothScanningMode.ACTIVE
+    )
+)
+```
+
+The below example shows registering to get callbacks for Nespresso Prodigios.
+
+```python
+from homeassistant.components import bluetooth
+
+...
+
+entry.async_on_unload(
+    bluetooth.async_register_callback(
+        hass, _async_nespresso_found, {"local_name": "Prodigio_*")}, bluetooth.BluetoothScanningMode.ACTIVE
+    )
+)
+```
+
+The below example shows registering to get callbacks for a device with the address `44:33:11:22:33:22`.
+
+```python
+from homeassistant.components import bluetooth
+
+...
+
+entry.async_on_unload(
+    bluetooth.async_register_callback(
+        hass, _async_specific_device_found, {"address": "44:33:11:22:33:22")}, bluetooth.BluetoothScanningMode.ACTIVE
+    )
+)
+```
+
+### Fetch the shared BleakScanner instance
+
+Integrations that need an instance of a `BleakScanner` should call the `bluetooth.async_get_scanner` API. This API returns a wrapper around a single `BleakScanner` that allows integrations to share without overloading the system.
+
+```python
+from homeassistant.components import bluetooth
+  
+scanner = bluetooth.async_get_scanner(hass)
+```
+
+
+### Determine if there is scanner running
```

Comment:
Determine if there is a scanner running

Reply:
Or: Determine if a scanner is running

Reply:
https://github.com/home-assistant/developers.home-assistant/pull/1455